mlir-sycl/lib/Conversion/SYCLToLLVM/DialectBuilder.cpp

```diff
@@ -76,24 +84,22 @@ ArrayAttr DialectBuilder::getI64ArrayAttr(ArrayRef<int64_t> vals) const {
 }
 
 //===----------------------------------------------------------------------===//
-// MemRefBuilder
+// FuncBuilder
 //===----------------------------------------------------------------------===//
 
-memref::AllocOp MemRefBuilder::genAlloc(MemRefType type) const {
-  return create<memref::AllocOp>(type);
-}
-
-memref::AllocaOp MemRefBuilder::genAlloca(MemRefType type) const {
-  return create<memref::AllocaOp>(type);
-}
-
-memref::CastOp MemRefBuilder::genCast(Value input,
-                                      MemRefType outputType) const {
-  return create<memref::CastOp>(outputType, input);
+func::CallOp FuncBuilder::genCall(FlatSymbolRefAttr funcSym, TypeRange resTypes,
+                                  ValueRange operands) const {
+  assert(funcSym && "Expecting a valid function symbol");
+  return create<func::CallOp>(resTypes, funcSym, operands);
 }
 
-memref::DeallocOp MemRefBuilder::genDealloc(Value val) const {
-  return create<memref::DeallocOp>(val);
+func::CallOp FuncBuilder::genCall(StringRef funcName, TypeRange resTypes,
+                                  ValueRange operands, ModuleOp &module) const {
+  assert(!funcName.contains('@') && "funcName should not contain '@'");
+  assert(module.lookupSymbol<LLVM::LLVMFuncOp>(funcName) &&
+         "Expecting to find a function declaration");
+  auto funcSym = SymbolRefAttr::get(builder.getContext(), funcName);
```

Comment:
Why not create func::CallOp directly with the StringRef? like
`return create<func::CallOp>(resTypes, funcName, operands);`

Reply:
Because I wanted to reuse the other ` FuncBuilder::genCall`.